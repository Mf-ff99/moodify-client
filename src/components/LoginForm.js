import React from 'react'
import AuthApiService from '../services/AuthApiService'
import TokenService from '../services/TokenService'

export default class LoginForm extends React.Component {


  state = {
    username: '',
    password: '',
    error: null
  }

  onLoginSuccess = () => {
    this.props.history.push('/home')
  }

  
  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    const { username, password } = ev.target
     
    AuthApiService.postLogin({
      user_name: username.value,
      password: password.value,
    })
      .then(res => {
        username.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.onLoginSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }


  render() {
    const { error } = this.state

    return (
      <section className="login-form">
        <h1>Log in</h1>
        <br />

        <form className="LoginForm"
         onSubmit={this.handleSubmitJwtAuth}
        >
          <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
          <div className="container">
            <label htmlFor="username"><b>Username </b></label>
            <input type="text" placeholder="Enter Username" name="username" defaultValue="dunder" required />
            <br />
            <br />
            <label htmlFor="password"><b>Password </b></label>
            <input type="password" placeholder="Enter Password" name="password" defaultValue="password" required />
            <br />
            <button type="submit" id="login-btn">Login</button>
            {/* <label>
        <input type="checkbox" checked="checked" name="remember"> Remember me</input>
      </label> */}
          </div>

          {/* <div class="container" style={{ "background-color": "#f1f1f1"}}>
      <button type="button" class="cancelbtn">Cancel</button>
      <span class="psw">Forgot <a href="#">password?</a></span>
    </div> */}
        </form>

      </section>
    )
  }
}