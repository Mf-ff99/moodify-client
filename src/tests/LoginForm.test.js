import React from 'react'
import LoginForm from '../components/LoginForm'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`LoginForm`, () => {
    it('Renders without crashing', () => {
        shallow(<LoginForm />)
    })
})