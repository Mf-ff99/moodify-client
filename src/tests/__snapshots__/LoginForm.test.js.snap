// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders form approriately 1`] = `
<section
  className="login-form"
>
  <h1>
    Log in
  </h1>
  <br />
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="container"
    >
      <label
        htmlFor="username"
      >
        <b>
          Username 
        </b>
      </label>
      <input
        defaultValue="dunder"
        name="username"
        placeholder="Enter Username"
        required={true}
        type="text"
      />
      <br />
      <br />
      <label
        htmlFor="password"
      >
        <b>
          Password 
        </b>
      </label>
      <input
        defaultValue="password"
        name="password"
        placeholder="Enter Password"
        required={true}
        type="password"
      />
      <br />
      <button
        id="login-btn"
        type="submit"
      >
        Login
      </button>
    </div>
  </form>
</section>
`;
